### ISSUE TITLE
Ability to vote for icons

### ISSUE THREAD

**Issue by x9136** [LawnchairLauncher/lawnicons#2107]
### About this discussion

I suggest adding the ability to vote for icons in the icon info sheet or somewhere else so that our users can easily express their opinion. There are many options for solving this issue, I'm talking about the preferred one. This will be useful, because **based on this info, you can redesign the icons so that Lawnicons is better suited to the needs of our users.**

How do I imagine the process: a user opens an icon somewhere → Tap 👍 or 👎  → We collect this info and evaluate what actions to take: replace, add another option, nothing... → After we make a change, the votes are reset.

This task should be solved after #1371, because when opinions are divided, it is better to leave a choice of several icons.

If this is implemented, and I have time by then, I will redo at least 100 icons, replacing the options that users did not like or adding additional ones.

**Comment by Hamster45105** [comment]
> There are many options for solving this issue

Here is an alternative that could be implemented now and used to decide on #2105:

An issue could be created on GitHub with details about the proposed change. People could then react with 👍 or 👎 within a set time period. In this case, I would suggest creating a new issue label called `icon-vote` or something similar. A button could then be put in the Lawnicons app that links to all issues with that label.

Alternatively, there is a discussion type on GitHub called [polls](https://github.blog/changelog/2022-04-12-discussions-polls/). If discussions were enabled, a poll could be created with details of the change.

These options would be much less complicated (in terms of code complexity) than having something built into the Lawnicons app. While of course this wouldn't reach all users, a notice could be posted in the Lawnchair Discord to reach more people. It is also important to take into account that people would need a GitHub account to vote.

Obviously, it would just be better to have multiple options for icons so that people can just choose what they want, so this wouldn't be needed if #1371 was implemented.

**Comment by x9136** [comment]
I have not considered creating issues or discussions for voting, because these options create too many unnecessary steps for the voter. You can compare it yourself if you evaluate the user's path when voting in Lawnicons and when voting on GitHub.

In addition, contributors will need to spend time organizing each vote and if there are a lot of it, then in the end it may take more time than it takes to develop voting in Lawnicons. Therefore, despite the ease of implementation, I assume that it will be even less effective than writing to our Discord with a request to vote.

My suggestion assumes that you can open Lawnicons (in the app or somewhere else) and vote for any icon at any time. This option would make it easy to cast a vote, collect statistics from all icons, and voters would be more representative, since the moment of voting is close to the moment of choosing a replacement for the icon they did not like.

**Comment by Hamster45105** [comment]
I think I misunderstood your original idea, thanks for clarifying.

I am curious though, how so many people found out to there was a vote in https://github.com/LawnchairLauncher/lawnicons/pull/1948#issuecomment-2029550425 ? Maybe a similar thing should be done for the Tesla icon in https://github.com/LawnchairLauncher/lawnicons/pull/2105

**Comment by x9136** [comment]
> how so many people found out there was a vote in https://github.com/LawnchairLauncher/lawnicons/pull/1948#issuecomment-2029550425 ?

The author of this PR is engaged in the development of Mihon. He dropped the link to their Discord and asked Mihon users to vote.

> Maybe a similar thing should be done for the Tesla icon in https://github.com/LawnchairLauncher/lawnicons/pull/2105

I believe the author went the other way.

**Comment by x9136** [comment]
After thinking about it for some more time, I decided that the opportunity to vote for icons is rarely needed and will be unclaimed. There are few icons for which you can add an option, and even fewer people who may need another version of some icon. The quality requirements are also as soft as possible, so I have not heard that users have asked to improve the quality.

Therefore, it would be a waste of time to do in-app voting.
